NSTTutorials/src/test/java/com/nst/tutorials/rest/runtimearguments/README.md

```diff
@@ -0,0 +1,73 @@
+
+### References
+- [RuntimeConfigManager](../../../../../../../../../NST/src/main/java/com/ebay/runtime/RuntimeConfigManager.java)
+
+## Adding custom runtime arguments
+
+To add a custom runtime argument that can be read during a test (for example), you can leverage the `RuntimeConfigManager` . 
+
+To create the runtime argument object, create a class that implements `RuntimeConfigValue`. All the runtime arguments above implement this interface, which you can reference as an example.
+
+[Here](RuntimeArgumentsCustomExample.java) is one such example.
+The generic defined with the `RuntimeConfigValue` interface defines what value that runtime argument should return.
+
+Once you have created your runtime argument class, use `RuntimeConfigManager.*getInstance*().addRuntimeArgument` to add it to the manager. Then, you can reference the value that is set during runtime by referencing the key, as so:
+
+`RuntimeConfigManager.*getInstance*().getRuntimeArgumentValue(CustomRuntimeArgument.***KEY***);`
+
+See the [associated test class](RuntimeArgumentsTest.java) for an example of adding a custom runtime argument.
+
+## Overriding an existing runtime argument
+
+We can override custom runtime arguments that have been set by retrieving the `RuntimeConfigValue` that has been set.
+Because the runtime argument values can be of any data type, we will need to retrieve and cast it using `RuntimeConfigManager.getRuntimeArgument`.
+
+```java
+RuntimeConfigValue<String> customRuntimeArgumentExampleValue =
+        (RuntimeConfigValue<String>) RuntimeConfigManager.getInstance().getRuntimeArgument(customRuntimeArgumentExample.getRuntimeArgumentKey());
+customRuntimeArgumentExampleValue.override("modifiedValue");
+```
+
+See the [associated test class](RuntimeArgumentsTest.java) for a runnable example that demonstrates this functionality.
+
+## Adding a custom logger
+
+NST supports customization of the logging / output that is generated. You may want to add your own custom implementation to modify the output of either the mock response files or the output to the UI test files as well.
+
+Custom loggers are **required** to be implemented if you want to export boilerplate code to UI test files, which is covered later on in detail in section 7.
+
+To do this, we will need to create our own custom logger file which implements *`FormatWriter`*:
+
+- `getPlatformAssociation` - This should return whichever platform you are creating a custom logger for (`IOS, ANDROID, MWEB, SITE`).
+- `writeMocks` - This is where you utilize the supplied list of service calls, along with the `testClassName` and `testMethodName` (if desired) to generate the mocks in the format of your choosing.
+- `updateTests` - This is where we can again utilize the list of service calls, along with the `testClassName` and `testMethodName` (if desired) to generate the UI test code output that is customized per your specifications.
```

Comment:
The following utility classes will be helpful: 

https://github.com/eBay/NSTSuite/blob/e6b51eb21725eafb46cce6a73dbf00d3fa9b1dfc/NST/src/main/java/com/ebay/service/logger/FormatWriterUtil.java

That utility may be useful.

For writing a UI test code generator you will want to introduce the concepts shown in our internal Android and iOS FUI writers (which are NOT part of this codebase) https://github.corp.ebay.com/eBayMobile/NST/blob/master/src/main/java/com/ebay/service/logger/platforms/processors/parser/AndroidJavaFuiTestParser.java

We cannot bring over eBay specific concepts, but the overarching concepts need to be conveyed.